lib/db_connection/connection_pool.ex

```diff
@@ -358,4 +378,23 @@ defmodule DBConnection.ConnectionPool do
     idle = :erlang.start_timer(timeout, self(), now_in_native, abs: true)
     %{codel | idle: idle}
   end
+
+  defp execute_queue_telemetry(status, queue, event, count \\ 1)
```

Comment:
I am not the biggest fun of executing those events and traversing the table on every operation, this can be quite expensive in the long term. I think a pull design will probably be a better fit here, as commented in the other PR. Basically, have an operation that reads the ETS the status and the ETS table on demand.

Reply:
Though this approach is quite dubious, there are a couple of thoughts that I've had:

1. It's a clearer API in comparison to finding a pool PID and then calling a function from an internal module
2. If the pool is ready, then there are either not that many queries to perform or checked in connection count is pretty low. In both cases an ets table scan is probably acceptable.
3. In case the pool is busy and the queue length is huge, the problem would be probably not the `ets.select_count` call(or it would be a much lesser problem). If the queue length is small, an ets scan wouldn't seem to be a problem.
4. Also, the approach with telemetry calls is easier to test

@josevalim What is your opinion?